src/oic/extension/client.py

```diff
@@ -340,9 +340,9 @@ def do_authorization_request(
 
     def store_registration_info(self, reginfo):
         self.registration_response = reginfo
-        self.client_secret = reginfo["client_secret"]
-        self.client_id = reginfo["client_id"]
-        self.redirect_uris = reginfo["redirect_uris"]
+        self.client_secret = reginfo.get("client_secret")
+        self.client_id = reginfo.get("client_id")
+        self.redirect_uris = reginfo.get("redirect_uris")
```

Comment:
This does clash with the definition of self.redirect_uris as a List[str]:

https://github.com/infohash/pyoidc/blob/b6da93d73001981e6afc5bf2e77f47e8e7889b44/src/oic/oauth2/__init__.py#L244